### PR TITLE
refactor: Convert analytics and dependencies to TypeScript

### PR DESCRIPTION
**Pull request by sjinks**
## Description

Related: https://github.com/Automattic/vip-cli/pull/1374

Converts src/lib/analytics and its dependencies (so that the configuration from #1377 does not find any issues) to TypeScript.

## Steps to Test

CI should pass.

### CODE REVIEW

File: __tests__/bin/vip-cache-purge-url.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ jest.mock( 'lib/api/cache-purge', () => ( {
 	purgeCache: jest.fn(),
 } ) );
 
-jest.mock( 'lib/tracker', () => ( {
+jest.mock( '../../src/lib/tracker', () => ( {
```
Comment (sjinks) [inline __tests__/bin/vip-cache-purge-url.js:28]: For Node.js 16, we have to use relative imports to TS files.

File: __tests__/bin/vip-config-envvar-delete.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ import { deleteEnvVarCommand } from '../../src/bin/vip-config-envvar-delete';
 import command from '../../src/lib/cli/command';
 import { deleteEnvVar, validateNameWithMessage } from '../../src/lib/envvar/api';
 import { cancel, confirm, promptForValue } from '../../src/lib/envvar/input';
+// $FlowExpectedError[cannot-resolve-module]
 import { trackEvent } from '../../src/lib/tracker';
```
Comment (sjinks) [inline __tests__/bin/vip-config-envvar-delete.js:17]: Flow cannot resolve TS modules; we must suppress such errors while converting the codebase.

File: __tests__/lib/analytics/index.js [hunk 3]
```diff
@@ -1,8 +1,18 @@
...
  * Internal dependencies
  */
 import Analytics from '../../../src/lib/analytics';
-import AnalyticsClientStub from '../../../src/lib/analytics/clients/stub';
+
+class AnalyticsClientStub {
+	trackEvent() {
+		return Promise.resolve( false );
+	}
+}
```
Comment (sjinks) [inline __tests__/lib/analytics/index.js:15]: We use this class only for tests. I removed it from the main source tree.

**Review by pschoffer — APPROVED** [review]
Nice

**Pull request merged**